### PR TITLE
chore: upgrade to tekton v0.56.4

### PR DESCRIPTION
**Pull request by l-qing**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

### CODE REVIEW

File: .build/build.yaml [hunk 1]
```diff
@@ -40,8 +40,7 @@ spec:
             - lines-coverage=10
             - passed-tests-rate=100
         - name: tool-image
-          # value: registry.alauda.cn:60080/devops/builder-go:1.18-ubuntu-495a3d88
-          value: build-harbor.alauda.cn/devops/builder-go:1.21-bullseye-9dcd2518
+          value: registry.alauda.cn:60080/devops/builder-go-121:latest
```
Comment (l-qing) [inline .build/build.yaml:43]: 切换到新的 golang 镜像中，以后也能跟着升级。

File: .build/build.yaml [hunk 2]
```diff
@@ -53,6 +52,7 @@ spec:
           workspace: cache
     ## ko publish image
     - name: build-entrypoint
+      retries: 1
```
Comment (l-qing) [inline .build/build.yaml:55]: 由于该流水线任务多，有时 会因为环境关系，导致单个 task 失败。
这里就加个重试，避免小概率失败，导致需要整体重试。

File: .build/build.yaml [hunk 3]
```diff
@@ -272,23 +279,41 @@ spec:
...
+            export GONOSUMDB=*
+
+            # modify th go.mod to fix vuln
+            go get github.com/docker/docker@v24.0.9
+            go get github.com/go-jose/go-jose/v3@v3.0.3
+            go get github.com/sigstore/cosign/v2@v2.2.4
+            go get golang.org/x/net@v0.23.0
+            go get google.golang.org/protobuf@v1.33.0
+            go get gopkg.in/go-jose/go-jose.v2@v2.6.3
+            go mod tidy
```
Comment (l-qing) [inline .build/build.yaml:307]: 修复 tkn 的应用漏洞。

File: .build/build.yaml [hunk 4]
```diff
@@ -416,6 +442,140 @@ spec:
...
+          workspace: cache
+      when:
+      - input: $(build.git.revision.type)
+        operator: in
+        values:
+        - "PullRequest"
+      params:
+        - name: targets
+          value:
+            - $(tasks.build-entrypoint.results.ociContainerImageBuild-url)
```
Comment (l-qing) [inline .build/build.yaml:468]: 增加 trivy 扫描。

**Review by danielfbm — APPROVED** [review]

**Pull request merged**